client/layout/guided-tours/steps.js

```diff
@@ -11,22 +11,40 @@ import Card from 'components/card';
 import Button from 'components/button';
 import ExternalLink from 'components/external-link';
 import Gridicon from 'components/gridicon';
-import { posToCss, getStepPosition, getBullseyePosition, targetForSlug } from './positioning';
+import { posToCss, getStepPosition, getBullseyePosition, getOverlayStyle, getScrollDiff, targetForSlug, getScrolledRect, query } from './positioning';
 
 class BasicStep extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = { stepPosition: null, targetRect: null };
+	}
+
+	componentWillMount() {
+		const container = query( '#secondary .sidebar' )[ 0 ];
+		const scrollY = getScrollDiff( this.props.targetSlug, container );
+		const targetRect = getScrolledRect( { targetSlug: this.props.targetSlug, scrollY: scrollY } );
+
+		this.setState( { stepPosition: posToCss( getStepPosition( this.props ) ) } );
+		this.setState( { targetRect: targetRect } );
```

Comment:
Actually, I'm not sure about the semantics of this whole block:

- `scrollY` is obtained via `positioning#getScrollDiff`, but then the value is only used in the call to `positioning#getScrolledRect`, the variable ending up GC'd after that. Does that mean that `getScrollDiff` is an implementation detail of `getScrolledRect`?
- `stepPosition` and `targetRect` are being saved as component state, but — since there is no other `setState` call for them in the component — they are more like instance constants. This is not super important and is more of a nuance in semantics, but it does mean that technically those two values could be stored as instance properties: `this.stepPosition = …`.
- Is `componentWillMount` the right place for this? Or, to put it differently: why can't we perform this calculation in `render`?